### PR TITLE
Fix/add create or update method to collection

### PR DESCRIPTION
**Pull request by ecoologic**
See Jira: https://zendesk.atlassian.net/browse/RED-887

This completes one of the open PRs and ensures it's tested properly.

Please ignore "live specs" being currently red, that's a different issue that will be addressed in a separate PR. Note the regular specs are green. I will merge when everything will be green anyway.

ORIGINAL PR: https://github.com/zendesk/zendesk_api_client_rb/pull/453

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -23,6 +22,7 @@ end
 
 group :dev do
   gem "bump"
+  gem "byebug"
```
Comment (ecoologic) [inline Gemfile:25]: I think it's time to remove the ruby 2.1 stuff.

File: lib/zendesk_api/collection.rb [hunk 2]
```diff
@@ -47,7 +47,7 @@ def initialize(client, resource, options = {})
     end
 
     # Methods that take a Hash argument
-    methods = %w{create find update update_many destroy}
+    methods = %w{create find update update_many destroy create_or_update}
```
Comment (ecoologic) [inline lib/zendesk_api/collection.rb:50]: this was the change from our external contributor

File: spec/core/collection_spec.rb [hunk 3]
```diff
@@ -49,6 +49,14 @@
...
     end
 
+    it "should defer #create_or_update to the resource class" do
+      resource = ZendeskAPI::Collection.new(client, ZendeskAPI::CreateOrUpdateTestResource)
+      stub_json_request(:post, %r{create_or_update_test_resources/create_or_update}, json(create_or_update_test_resource: { param: "abc" }))
+      resource.create_or_update!
+
+      assert_requested(:post, %r{create_or_update_test_resources/create_or_update$})
+    end
+
```
Comment (ecoologic) [inline spec/core/collection_spec.rb:59]: ... and the spec now is actually testing the request, similarly to other tests

**Review by tagpro — APPROVED** [review]
LGTM

**Pull request merged**